website/docs/plugin/framework/functions/errors.mdx

```diff
@@ -0,0 +1,149 @@
+---
+page_title: 'Plugin Development - Framework: Function Errors'
+description: |-
+  How to return function errors from the Terraform provider development
+  framework.
+---
+
+# Returning Function Errors
+
+Providers use [`FuncError`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#FuncError) to
+surface a practitioner-facing error generated during execution of provider-defined functions. These errors are
+returned from Terraform CLI at the end of command output:
+
+```console
+$ terraform plan
+# ... other plan output ...
+╷
+│ Error: Message
+│ 
+│   on example.tf line #:
+│    #: source configuration line
+╵
```

Comment:
FYI the error diagnostic for a regular function error appears to be:

```
        Error: Error in function call
        
          on example.tf line #:
          #: source configuration line
        
        Call to function "{FUNCTION}" failed: {TEXT}.
```

And for an argument error:

```
        Error: Invalid function argument
        
          on example.tf line #:
          #: source configuration line
        
        Invalid value for "{POSITION}" parameter: {TEXT}.
```

Reply:
Thanks. Have updated the docs with an example of both.